packages/usbmux/usbmux.0.9/opam

```diff
@@ -0,0 +1,48 @@
+opam-version: "1.2"
+maintainer: "Edgar Aroutiounian <edgar.factorial@gmail.com>"
+authors: "Edgar Aroutiounian <edgar.factorial@gmail.com>"
+homepage: "https://github.com/onlinemediagroup/ocaml-usbmux"
+bug-reports: "https://github.com/onlinemediagroup/ocaml-usbmux/issues"
+license: "BSD-3"
+dev-repo: "https://github.com/onlinemediagroup/ocaml-usbmux.git"
+build: [
+  ["ocaml" "setup.ml" "-configure" "--prefix" prefix]
+  ["ocaml" "setup.ml" "-build"]
+]
+install: ["ocaml" "setup.ml" "-install"]
+build-test: [
+  ["ocaml" "setup.ml" "-configure" "--enable-tests"]
+  ["ocaml" "setup.ml" "-build"]
+  ["ocaml" "setup.ml" "-test"]
+]
+remove: ["ocaml" "%{etc}%/usbmux/_oasis_remove_.ml" "%{etc}%/usbmux"]
+depends: [
+  "ANSITerminal"
+  "cmdliner" {build}
```

Comment:
> oasis2opam makes that decision because cmdliner is the dependency of an executable only.

This is a correct interpretation of what the opam's development manual says, but I'd argue that the manual is wrong.  `{build}` dependencies should only be about things your build system needs in order to be run. It should not be about dependencies of the artefacts you build.

> On the other hand, for an executable, one may argue that its behavior has been tested — and dependencies issues have been worked around — and that an update of the library may cause undesired behavior...

This is not a good argument since if you restart from scratch you will then build the binary with the latest version of the library which will cause the possible undesired behaviour.

In general it is better if we try to maximise the reproducibility of builds and minimize `opam`'s upgrade history on the outcomes of a given install. W.r.t. `oasis2opam`'s behaviour does not enable this.